src/test/java/seedu/address/logic/commands/ResumePreviewCommandTest.java

```diff
@@ -0,0 +1,118 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_ITEM;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FOURTH_ITEM;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_ITEM;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.results.ResumePreviewCommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.item.Internship;
+import seedu.address.model.item.Person;
+import seedu.address.model.item.Project;
+import seedu.address.model.item.Resume;
+import seedu.address.model.item.Skill;
+import seedu.address.testutil.TypicalInternship;
+import seedu.address.testutil.TypicalProject;
+import seedu.address.testutil.TypicalResumeBook;
+import seedu.address.testutil.TypicalSkill;
+
+public class ResumePreviewCommandTest {
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+    }
+
+    @Test
+    public void execute_invalidResumeIndex_throwsCommandException() {
+        Index invalidIndex = INDEX_FOURTH_ITEM;
+        ResumePreviewCommand resumePreviewCommand = new ResumePreviewCommand(invalidIndex);
+        assertThrows(CommandException.class,
+                Messages.MESSAGE_INVALID_INDEX, () -> resumePreviewCommand.execute(model));
+    }
+
+    @Test
+    public void execute_emptyResume_success() {
+        Model expectedModel = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+        Index validIndex = INDEX_FIRST_ITEM;
+        ResumePreviewCommand resumePreviewCommand = new ResumePreviewCommand(validIndex);
+        String data = createData(model.getUser(), model.getResumeByIndex(Index.fromOneBased(1)), new ArrayList<>(),
```

Comment:
```suggestion
        String data = createData(model.getUser(), model.getResumeByIndex(INDEX_FIRST_ITEM), new ArrayList<>(),
```

Would this be better instead? Or perhaps using `validIndex` even